devnotes/isteps/14_DRAM_init/1_mss_memdiag.md

```diff
@@ -0,0 +1,633 @@
+    .template writeBit<PT::RMW_WRT_AUTOINC>(mss::states::ON)
+    .template writeBit<PT::RMW_WRT_ECCGEN>(mss::states::ON);
+
+    // This loop will be run twice to write the pattern twice.  Once per 64B write.
+    // When MCBIST maint mode is in 64B mode it will only use the first 64B when in 128B mode
+    // MCBIST maint will use all 128B (it will perform two consecutive writes)
+    const auto l_ports =  mss::find_targets<TT::PORT_TYPE>(i_target);
+    // Init the port map
+    for (const auto& p : l_ports)
+    {
+        l_aacr.template insertFromRight<PT::RMW_WRT_ADDRESS, PT::RMW_WRT_ADDRESS_LEN>(PT::MAINT_DATA_INDEX_START);
+        for (auto l_num_writes = 0; l_num_writes < 2; ++l_num_writes)
+        {
+            fapi2::putScom(p, PT::RMW_WRT_BUF_CTL_REG, l_aacr);
+            for (const auto& l_cache_line : i_pattern)
+            {
+                fapi2::buffer<uint64_t> l_value_first  = i_invert ? ~l_cache_line.first : l_cache_line.first;
+                fapi2::buffer<uint64_t> l_value_second = i_invert ? ~l_cache_line.second : l_cache_line.second;
+                fapi2::putScom(p, PT::RMW_WRT_BUF_DATA_REG, l_value_first);
+                // In order for the data to actually be written into the RMW buffer, we must issue a putscom to the MCA_AAER register
+                // This register is used for the ECC, we will just write all zero to this register.  The ECC will be auto generated
+                // when the aacr MCA_WREITE_AACR_ECCGEN bit is set
+                fapi2::putScom(p, PT::RMW_WRT_BUF_ECC_REG, 0);
+                // No need to increment the address because the logic does it automatically when MCA_WREITE_AACR_AUTOINC is set
+                fapi2::putScom(p, PT::RMW_WRT_BUF_DATA_REG, l_value_second);
+                // In order for the data to actually be written into the RMW buffer, we must issue a putscom to the MCA_AAER register
+                // This register is used for the ECC, we will just write all zero to this register.  The ECC will be auto generated
+                // when the aacr MCA_WREITE_AACR_ECCGEN bit is set
+                fapi2::putScom(p, PT::RMW_WRT_BUF_ECC_REG, 0);
+            }
+            l_aacr.template insertFromRight<PT::RMW_WRT_ADDRESS, PT::RMW_WRT_ADDRESS_LEN>(PT::MAINT_DATA_INDEX_END);
+        }
+    }
+}
+
+template< mss::mc_type MC = DEFAULT_MC_TYPE, fapi2::TargetType T, typename TT = mcbistTraits<MC, T> >
+inline fapi2::ReturnCode load_config( const fapi2::Target<T>& i_target, const mcbist::program<MC>& i_program )
+{
+    // Copy the program's config settings - we want to modify them if we're in sim.
+    fapi2::buffer<uint64_t> l_config = i_program.iv_config;
+    // If we're running in Cronus, there is no interrupt so any attention bits will
+    // hang something somewhere. Make sure there's nothing in this config which can
+    // turn on attention bits unless we're running in hostboot
+#ifndef __HOSTBOOT_MODULE
```

Comment:
__HOSTBOOT_MODULE is always defined for us, so it may be skipped.

Reply:
I have removed it